prepare_data/src/test/groovy/org/orbisgis/common/InputDataFormattingTest.groovy

```diff
@@ -762,6 +848,20 @@ class InputDataFormattingTest {
         assertEquals('high', h2GISDatabase.firstRow("SELECT HEIGHT_CLASS FROM VEGET " +
                 "WHERE ID_SOURCE='ZONEVEGE0000000222262077';")["HEIGHT_CLASS"])
 
+        // Check if vegetation area are well selected or not ...
+        // ... with the veget area 'ZONEVEGE0000000222259852' which is inside the zone --> so expected 1
+        assertEquals(1, h2GISDatabase.firstRow("SELECT COUNT(*) as TOTAL FROM VEGET " +
+                "WHERE ID_SOURCE='ZONEVEGE0000000222259852';")["TOTAL"])
+        // ... with the veget area 'ZONEVEGE0000000222246505' which is inside the extended zone --> so expected 1
+        assertEquals(1, h2GISDatabase.firstRow("SELECT COUNT(*) as TOTAL FROM VEGET " +
+                "WHERE ID_SOURCE='ZONEVEGE0000000222246505';")["TOTAL"])
+        // ... with the veget area 'ZONEVEGE0000000222257768' which is intersecting the extended zone (having a part outside) --> so expected 1
```

Comment:
for hydro and vegetation we merge geometries that are in contact and then could keep some of the resulting geometries that are splitted by the zone geometry depending on the vegetation (or hydro) total size. Is it possible to also test that ?

Reply:
We will indeed have to test this, but in my opinion it should not be done here, but rather when the RSUs are created.
Indeed, the cumulative area criterion (for objects that touch) is only used for RSUs.
Here we select the water and vegetation surfaces on a larger area, which has been made to select those objects that could be far from the municipality but still need to be taken into account.

Reply:
You are right here we keep it without any condition of size. Only in the RSU creation we merge and then test their size.